### PR TITLE
Ports anomaly containers from Bay

### PR DESCRIPTION
**Pull request by jknpj**
anomaly
**A safe way to transport deadly shit.**

Finally a safe way for traders sell artifacts and xenoarchs to take their research to centcomm. Right now they can only store unknown artifacts, but they're pretty much the only type of artifact that need this kind of container anyway.

:cl: 
 * rscadd: Adds anomaly container, a safe way to transport and store unknown artifacts. Avaible at cargo for 50 zeny.

### CODE REVIEW

File: code/datums/supplypacks.dm [hunk 1]
```diff
@@ -1641,6 +1641,15 @@ var/list/all_supply_groups = list("Supplies","Clothing","Security","Hospitality"
 	access = access_xenobiology
 	group = "Science"
 
+/datum/supply_packs/anomaly_container
+	name = "Anomaly container"
+	contains = list (/obj/structure/anomaly_container)
+	cost = 50
+	containertype = /obj/structure/closet/crate/secure/scisec
```
Comment (ghost) [inline code/datums/supplypacks.dm]: So it's a crate, locked inside a smaller secure crate?

Comment (Shadowmech88) [reply code/datums/supplypacks.dm]: It'd make more sense for this kind of thing to be shipped in a big wooden box like what cows come in, wouldn't it?

**Comment by SonixApache** [comment]
>50 zenny

fucking capcom pricing

**Comment by Kurfursten** [comment]
But can it safely contain a damaged supermatter shard?

The containers shards come in don't take them back once they've been destabilized at all

**Comment by PJB3005** [comment]
> But can it safely contain a damaged supermatter shard?

Why should it?

File: code/modules/research/xenoarchaeology/tools/anomaly_container.dm [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	density = 1
+
+	var/obj/machinery/artifact/contained
+
+/obj/structure/anomaly_container/attack_hand(var/mob/user)
+	if(!contained)
+		return
+	if(alert(user, "Do you wish release \the [contained] from \the [src]?", "Confirm", "Yes", "No") != "Yes")
+		return
+	if(Adjacent(user))
```
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/tools/anomaly_container.dm]: Needs to check `user.incapacitated` too.

File: code/modules/research/xenoarchaeology/tools/anomaly_container.dm [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		release()
+
+/obj/structure/anomaly_container/attack_robot(var/mob/user)
```
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/tools/anomaly_container.dm]: Should just make this call `attack_hand` outright probably.

**Comment by Kurfursten** [comment]
>A safe way to transport deadly shit

File: code/modules/research/xenoarchaeology/tools/anomaly_container.dm [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+/obj/structure/anomaly_container/proc/contain(var/obj/machinery/artifact/artifact)
+	if(contained)
+		return
+	contained = artifact
+	artifact.forceMove(src)
+	underlays += image(artifact)
+	desc = "Used to safely contain and move anomalies. \The [contained] is kept inside."
+
+/obj/structure/anomaly_container/proc/release()
```
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/tools/anomaly_container.dm:28]: This could probably use logging.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by jknpj** [comment]
>But can it safely contain a damaged supermatter shard?

It only contain unknown artifacts.

**Comment by Kurfursten** [comment]
Well that's okay. It'd probably end up more desirable for engineering than science if it could.

**Comment by jknpj** [comment]
We need those black supermatter containers to be ordeable too.

**Comment by PJB3005** [comment]
Order a new shard?

**Comment by Sprok0** [comment]
Those don't come in a shard box, just a normal crate. The special black shard crates are unique and can't be replaced.

File: code/modules/research/xenoarchaeology/tools/anomaly_container.dm [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	density = 1
+
+	var/obj/machinery/artifact/contained
+
+/obj/structure/anomaly_container/attack_hand(var/mob/user)
+	if(!contained || user.incapacitated() || user.lying)
+		return
+	if(alert(user, "Do you wish release \the [contained] from \the [src]?", "Confirm", "Yes", "No") != "Yes")
+		return
+	if(Adjacent(user))
```
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/tools/anomaly_container.dm]: Still need to check for alive-ness here.

Comment (jknpj) [reply code/modules/research/xenoarchaeology/tools/anomaly_container.dm]: aren't ghosts incapacited?

Comment (PJB3005) [reply code/modules/research/xenoarchaeology/tools/anomaly_container.dm]: The user could get killed after opening the alert window and still he adjacent.

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**